esp-hal/src/soc/esp32s2/regi2c.rs

```diff
@@ -34,23 +38,180 @@ define_regi2c! {
+        }
+        reg: I2C_BBPLL_REG5(5) {
+            field: I2C_BBPLL_OC_DR1(2..0),
+            field: I2C_BBPLL_OC_DR3(6..4),
+            field: I2C_BBPLL_EN_USB(7..7)
+        }
+        reg: I2C_BBPLL_REG6(6) {
+            field: I2C_BBPLL_OC_DCUR(2..0),
+            field: I2C_BBPLL_INC_CUR(3..3),
+            field: I2C_BBPLL_OC_DHREF_SEL(5..4),
+            field: I2C_BBPLL_OC_DLREF_SEL(7..6)
+        }
+        reg: I2C_BBPLL_REG8(8) {
+            field: I2C_BBPLL_OR_CAL_CAP(3..0),
+            field: I2C_BBPLL_OR_CAL_UDF(4..4),
+            field: I2C_BBPLL_OR_CAL_OVF(5..5),
+            field: I2C_BBPLL_OR_CAL_END(6..6),
+            field: I2C_BBPLL_OR_LOCK(7..7)
+        }
+        reg: I2C_BBPLL_REG9(9) {
+            field: I2C_BBPLL_BBADC_DELAY2(3..2),
+            field: I2C_BBPLL_BBADC_DVDD(5..4),
+            field: I2C_BBPLL_BBADC_DREF(7..6)
+        }
+        reg: I2C_BBPLL_REG10(10) {
+            field: I2C_BBPLL_BBADC_DCUR(1..0),
+            field: I2C_BBPLL_BBADC_INPUT_SHORT(2..2),
+            field: I2C_BBPLL_ENT_PLL(3..3),
+            field: I2C_BBPLL_DTEST(5..4),
+            field: I2C_BBPLL_ENT_ADC(7..6)
+        }
+    }
+    master: REGI2C_APLL(0x6D, 1) {
+        reg: I2C_APLL_REG0(0) {
+            field: I2C_APLL_IR_CAL_DELAY(3..0),
+            field: I2C_APLL_IR_CAL_RSTB(4..4),
+            field: I2C_APLL_IR_CAL_START(5..5),
+            field: I2C_APLL_IR_CAL_UNSTOP(6..6),
+            field: I2C_APLL_OC_ENB_FCAL(7..7)
+        }
+        reg: I2C_APLL_REG1(1) {
+            field: I2C_APLL_IR_CAL_EXT_CAP(4..0),
+            field: I2C_APLL_IR_CAL_ENX_CAP(5..5),
+            field: I2C_APLL_OC_LBW(6..6)
+        }
+        reg: I2C_APLL_REG2(2) {
+            field: I2C_APLL_IR_CAL_CK_DIV(3..0),
+            field: I2C_APLL_OC_DCHGP(6..4),
+            field: I2C_APLL_OC_ENB_VCON(7..7)
+        }
+        reg: I2C_APLL_REG3(3) {
+            field: I2C_APLL_OR_CAL_CAP(4..0),
+            field: I2C_APLL_OR_CAL_UDF(5..5),
+            field: I2C_APLL_OR_CAL_OVF(6..6),
+            field: I2C_APLL_OR_CAL_END(7..7)
+        }
+        reg: I2C_APLL_REG4(4) {
+            field: I2C_APLL_OR_OUTPUT_DIV(4..0),
+            field: I2C_APLL_OC_TSCHGP(6..6),
+            field: I2C_APLL_EN_FAST_CAL(7..7)
+        }
+        reg: I2C_APLL_REG5(5) {
+            field: I2C_APLL_OC_DHREF_SEL(1..0),
+            field: I2C_APLL_OC_DLREF_SEL(3..2),
+            field: I2C_APLL_SDM_DITHER(4..4),
+            field: I2C_APLL_SDM_STOP(5..5),
+            field: I2C_APLL_SDM_RSTB(6..6)
+        }
+        reg: I2C_APLL_REG6(6) {
+            field: I2C_APLL_OC_DVDD(4..0)
+        }
+        reg: I2C_APLL_REG7(7) {
+            field: I2C_APLL_DSDM2(5..0)
+        }
+        reg: I2C_APLL_REG8(8) {
+            field: I2C_APLL_DSDM1(7..0)
+        }
+        reg: I2C_APLL_REG9(9) {
+            field: I2C_APLL_DSDM0(7..0)
+        }
+    }
 }
+#[ram]
+pub unsafe fn i2c_rtc_enable_block(block: u8) {
```

Comment:
I don't think these should be `#[ram]`, they aren't really performance critical. regi2c is mostly just used in initialization code right now.

Reply:
This neither has to be public, nor unsafe, either.